common/common-rest/src/main/java/org/apache/servicecomb/common/rest/codec/param/PathProcessorCreator.java

```diff
@@ -52,6 +54,9 @@ public Object getValue(HttpServletRequest request) {
       if (value == null) {
         return null;
       }
+      if (value.contains(":")) {
+        return convertValue(URLDecoder.decode(value, "UTF-8"), targetType);
```

Comment:
I think this modification is not correct. You can add a test case for HttpUtils, like 
```
  @Test
  public void uriEncode_sss() {
    String encoded = HttpUtils.uriEncodePath("a:b");
    Assert.assertEquals("a:b", encoded);
    Assert.assertEquals("a:b", HttpUtils.uriDecodePath(encoded));
  }

```

And make this case pass.

Reply:
I notic there is a case org.apache.servicecomb.foundation.common.http.TestHttpUtils#uriDecode_failed
to show a error when the param of HttpUtils.uriDecodePath contain ":".

I think modify  the HttpUtils.uriDecodePat will change the previous  logic.

Reply:
Maybe this test case is not correct. Checked the [RFC](https://www.ietf.org/rfc/rfc2396.txt) 

```
3.3. Path Component

   The path component contains data, specific to the authority (or the
   scheme if there is no authority component), identifying the resource
   within the scope of that scheme and authority.

      path          = [ abs_path | opaque_part ]

      path_segments = segment *( "/" segment )
      segment       = *pchar *( ";" param )
      param         = *pchar

      pchar         = unreserved | escaped |
                      ":" | "@" | "&" | "=" | "+" | "$" | ","

   The path may consist of a sequence of path segments separated by a
   single slash "/" character.  Within a path segment, the characters
   "/", ";", "=", and "?" are reserved.  Each path segment may include a
   sequence of parameters, indicated by the semicolon ";" character.
   The parameters are not significant to the parsing of relative
   references.
```

Reply:
I checked the code again. May be "encodePathParam" and "decodePathParam" should be used here. You need check details usage of "uriEncodePath" and "uriDecodePath". They are different.

Reply:
Add decodePathParam method.